### DISCUSSION TITLE
Getting focal length from image (Canon compressed CR3)

### DISCUSSION THREAD

**Discussion by kkfin** [ImageMagick/ImageMagick#7128] [Help] [answered]
### ImageMagick version

7.1.1-28 Q16-HDRI x64 365d9b6:20240211

### Operating system, version and so on

Windows 11

### Description

I'm trying to get the Focal Length info from a Canon CR3 raw file with command 
`magick <path to image> %[EXIF:FocalLength]`
It does not work but instead returns the following:
```
magick: no encode delegate for this image format `CR3' @ warning/constitute.c/WriteImage/1400.
magick: no encode delegate for this image format `CR3' @ error/constitute.c/WriteImage/1409.
```

However if I try to transform the image with command 
`magick <path to image> -set filename:f "%%[d]/%%[t]" -resize "%%[fx:w*1.6]"x"%%[fx:h]!" "%%[filename:f].jpg"`
 it can produce a JPG from it.

Command `magick -version` returns
```
Version: ImageMagick 7.1.1-28 Q16-HDRI x64 365d9b6:20240211 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Channel-masks(64-bit) Cipher DPC HDRI Modules OpenCL OpenMP(2.0)
Delegates (built-in): bzlib cairo flif freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193732826)
```

Also command `magick identify -list format` returns among other lines the following `CR3  DNG       r--   Canon Digital Camera Raw Image Format (0.21.2-Release)` so Canon RAW CR3 files should be fully supported. And indeed the transforming CR3 -> JPG works, it's the EXIF data retrieving that is not working.

Imagemagick is installed using the official binary installer for windows (https://imagemagick.org/script/download.php#windows).

I'm using CRAW setting in my camera (compressed raw), could this be the reason EXIF data can't be retrieved? 

As a background info, I'm trying to build a batch file where images are identified based on their focal length. I'm shooting with regular Canon RF lenses and third party anamorphic lens. The anamorphic images can be recognized by their focal length info as it's recorded as 0 mm (as there is no electric contacts between lens and camera body).

**Comment by snibgo** [comment, marked as answer]
> magick <path to image> %[EXIF:FocalLength]

That command is wrong. It should end in a filename, or pseudo-file such as `info:` for text data.

Perhaps exif data is not available to IM. This command ...
```
magick yourfile.cr3 -verbose info:
```
... will tell you what metadata is available. For a file on my computer, I can get `dng:focal.length`, so:
```
C:\web\im>%IMG7%magick myfile.CR2 -format %[dng:focal.length] info:
7e+01 mm
```

As a general rule, if you only need metadata and not actual images, exiftool may be a more suitable tool than ImageMagick.
```
C:\web\im>exiftool -focallength myfile.CR2
Focal Length                    : 70.0 mm
```

**Reply by kkfin** [reply]
I know about **exiftool** and it can show the focal length from the image. However, I think I can't use it in this case as it is not recommended to be used in loops (see https://exiftool.org/mistakes.html#M3). I might have quite a lot of images to process (for example currently I have a little bit over 400 images of which about 280 are anamorphic) so using exiftool in loop to process these could be quite slow and/or resource hungry.

However, command you suggested `magick yourfile.cr3 -verbose info:` revealed that indeed the EXIF -tag is not correct but `%[dng:focal.length]` is. The command `magic <path to image> -format %[dng:focal.length] info:` returns `0 mm` which is the info I'm looking for.

Thanks a lot :)
